### ISSUE TITLE
Change the column type from string to jsonb and cast the previous value to a new json

### ISSUE THREAD

**Issue by redBaron23** [knex/knex#5336]
For example I've a column that's name (type string) and I'd like to change the type to jsonb, and keep the news values like this
```
{ first_name: PREVIOUS_NAME_COLUMN, last_name: "" }
```

Do anyone had this issue before?

**Comment by JhonatanZubietaMooveIt** [comment]
Yes, I have the same problem, we have a varchar column and we want to change the type to jsonb, the tool migration generated this script: 

```
knex.schema.alterTable('table_name', table => {
   table.jsonb('column_name').alter();
});
```

And we are receiving this message error:
`alter table "table_name" alter column "column_name" type jsonb using ("column_name"::jsonb) - invalid input syntax for type json`

For this kinda change we should use `using to_jsonb(message);`  but how can we add this sentence to the script?

**Comment by mercmobily** [comment]
Hi! Thanks for the request. Changing a column from string to `jsonb` and casting previous values is supported via raw SQL in PostgreSQL; Knex doesn’t currently offer a single builder method to perform the cast, but you can express it cleanly in a migration.

Example up migration:

```js
// Drop any defaults/constraints referring to the old type first if needed
// Option A: values already contain valid JSON text... parse it
await knex.raw('alter table ?? alter column ?? type jsonb using ??::jsonb', [
  'your_table',
  'your_column',
  'your_column',
]);
```

Knex will parameterise identifiers safely, and the `USING` clause applies the server‑side conversion.

Thanks @JhonatanZubietaMooveIt for pointing out the common case where existing data is arbitrary text rather than JSON. If you want to preserve the original text as JSON strings (and avoid cast failures on non‑JSON values), use `to_jsonb(column)` instead of a direct cast:

```js
// Option B: values are arbitrary text... store them as JSON strings
await knex.raw('alter table ?? alter column ?? type jsonb using to_jsonb(??)', [
  'your_table',
  'your_column',
  'your_column',
]);
```

If some rows contain valid JSON and others do not, PostgreSQL does not offer a "try cast" in plain SQL. In that case, either clean the data ahead of time, run the change in batches, or choose a single behaviour (parse vs wrap as string) that fits your application.

We can consider an enhancement to the PG schema compiler, but raw DDL is the standard approach for type changes that require a `USING` clause like the above.
